consensus/ledger_validator/ledger_validator.c

```diff
@@ -85,44 +88,58 @@ static retcode_t build_snapshot(ledger_validator_t *const lv,
                ", Candidate: #%" PRIu64 "\n",
                *consistent_index, milestone.index);
     }
-    // TODO state diff model
-    //       StateDiffViewModel stateDiffViewModel =
-    //           StateDiffViewModel.load(tangle, candidateMilestone.getHash());
-    //
-    //       if (stateDiffViewModel != null && !stateDiffViewModel.isEmpty()) {
-    //         if
-    //         (Snapshot.isConsistent(milestoneTracker.latestSnapshot.patchedDiff(
-    //                 stateDiffViewModel.getDiff()))) {
-    //           milestoneTracker.latestSnapshot.apply(stateDiffViewModel.getDiff(),
-    //                                                 candidateMilestone.index());
-    *consistent_index = milestone.index;
-    memcpy(consistent_hash, milestone.hash, FLEX_TRIT_SIZE_243);
-    //         } else {
-    //           break;
-    //         }
-    //       }
+    if ((ret = iota_tangle_state_delta_load(lv->tangle, milestone.index,
+                                            &delta)) != RC_OK) {
+      goto done;
+    }
+    if (delta != NULL && !state_delta_empty(delta)) {
+      if ((ret = iota_snapshot_create_patch(
+               lv->milestone_tracker->latest_snapshot, &delta, &patch)) !=
+          RC_OK) {
+        goto done;
+      }
+      if (state_delta_is_consistent(&patch)) {
```

Comment:
don't we want to return error/log when patch is not consistent?

Reply:
We should definitely log something.
I'm not sure if we should return an error though, I'll investigate.

Reply:
Yes, definitely an error, done.